integration_test.go

```diff
@@ -0,0 +1,231 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"math/rand"
+	"os"
+	"os/exec"
+	"testing"
+	"text/template"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/api/iterator"
+)
+
+const (
+	logFilePath    = "./logfile.log"
+	logFileName    = "logfile.log"
+	configFilePath = "./fluent-bit.conf"
+)
+
+// integration test validates the end-to-end fluentbit and bigquery pipeline
+func TestPipeline(t *testing.T) {
+
+	ctx := context.Background()
+
+	//get projectID from environment
+	projectID := os.Getenv("ProjectID")
+	if projectID == "" {
+		t.Fatal("Environment variable 'ProjectID' is required to run this test, but not set currently")
+	}
+
+	// Set up BigQuery client
+	client, err := bigquery.NewClient(ctx, projectID)
+	if err != nil {
+		t.Fatalf("Failed to create BigQuery client: %v", err)
+	}
+	defer client.Close()
+
+	// Create a random dataset and table name
+	datasetHash := randString()
+	datasetID := "testdataset_" + datasetHash
+
+	tableHash := randString()
+	tableID := "testtable_" + tableHash
+
+	// Create BigQuery dataset and table in an existing project
+	tableSchema := bigquery.Schema{
+		{Name: "Message", Type: bigquery.StringFieldType},
+	}
+	dataset := client.Dataset(datasetID)
+	if err := dataset.Create(ctx, &bigquery.DatasetMetadata{Location: "US"}); err != nil {
+		t.Fatalf("Failed to create BigQuery dataset %v", err)
+	}
+	table := dataset.Table(tableID)
+	if err := table.Create(ctx, &bigquery.TableMetadata{Schema: tableSchema}); err != nil {
+		t.Fatalf("Failed to create BigQuery table: %v", err)
+	}
+
+	//Create config file with random table name
+	if err := createConfigFile(projectID, datasetID, tableID); err != nil {
+		t.Fatalf("failed to create config file: %v", err)
+	}
+
+	// Create log file
+	file, err := os.Create(logFilePath)
+	if err != nil {
+		t.Fatalf("Failed to create log file: %v", err)
+	}
+	defer file.Close()
+
+	// Start Fluent Bit with the config file
+	FBcmd := exec.Command("fluent-bit", "-c", "fluent-bit.conf")
+	if err := FBcmd.Start(); err != nil {
+		t.Fatalf("Failed to start Fluent Bit: %v", err)
+	}
+
+	//Wait for fluent-bit connection to generate data; add delays before ending fluent-bit process
+	time.Sleep(2 * time.Second)
+	if err := generateData(10); err != nil {
```

Comment:
Can we define 10 as a constant and use it during verification?

Reply:
Done